talk/objectorientation/objectsclasses.tex

```diff
@@ -65,37 +65,58 @@
 
 \begin{frame}[fragile]
   \frametitlecpp[98]{Separating the interface}
-  \begin{block}{Header : MyFirstClass.hpp}
-    \begin{cppcode*}{}
-      #pragma once
-      struct MyFirstClass {
-        int a;
-        void squareA();
-        int sum(int b);
-      };
-    \end{cppcode*}
-  \end{block}
-  \begin{block}{Implementation : MyFirstClass.cpp}
-    \begin{cppcode*}{}
-      #include "MyFirstClass.hpp"
-      void MyFirstClass::squareA() {
-        a *= a;
-      }
-      int MyFirstClass::sum(int b) {
-        return a + b;
-      }
-    \end{cppcode*}
-  \end{block}
+  \begin{columns}[t]
+    \begin{column}{.45\textwidth}
+    \begin{block}{Header: MyClass.hpp}
+      \begin{cppcode*}{gobble=4}
+        #pragma once
+        struct MyClass {
+          int a;
+          void squareA();
+        };
+      \end{cppcode*}
+    \end{block}
+    \begin{block}{Implementation: MyClass.cpp}
+      \begin{cppcode*}{gobble=4}
+        #include "MyClass.hpp"
+        void MyClass::squareA() {
+          a *= a;
+        }
+      \end{cppcode*}
+    \end{block}
+    \end{column}
+    \begin{column}{.45\textwidth}
+    \begin{block}{User 1: main.cpp}
+      \begin{cppcode*}{gobble=4}
+        #include "MyClass.hpp"
+        int main() {
+          MyClass mc;
+          ...
+        }
+      \end{cppcode*}
+    \end{block}
+    \begin{block}{User 2: fun.cpp}
+      \begin{cppcode*}{gobble=4}
+        #include "MyClass.hpp"
+        int fun(MyClass& mc) {
+          return mc.squareA();
```

Comment:
The function returns `void`, so maybe it's better to make it return something.

Reply:
Fixed, made everything return `void`.